### PR TITLE
[PD-3847] Add Delete to Partner Assessments

### PR DESCRIPTION
**Pull request by koonsies**
## What's this PR do?
Adds delete functionality to partner assessments

## Where should the reviewer start?
Partner Assessments

## How should this be manually tested?
Create an assessment
Edit that assessment
Click "delete"
Confirm
See that it doesn't show up in the list (it's actually archived, not deleted)

## Any background context you want to provide?
This relies on the ubuntu:xenial changes in another PR. 

### Front-End

- [ ] Works on IE 11 and Edge
- [ ] Works on Mobile
- [ ] Unit tests added or ticket exists to add tests

### Separation of Concerns

- [ ] Back end "business logic" is implemented in plain python, Django model classes, and Django forms. (not Django views or other frameworks)
- [ ] Concerns of UI, api, and model are in separate code units.

### Tests

- [ ] Base classes used for python tests don't do too much unnecessary work.

### Migrations

- [ ] Migrations can be reversed.
- [ ] Current production code would continue to work after this migration is applied.
- [ ] Data migrations' forward/reverse functions should be decorated with `@only_on_default`.

### CODE REVIEW

**Comment by koonsies** [comment]
Closing until I can add tests

File: gulp/src/partnerassessment/components/Form/AssessmentForm.jsx [hunk 1]
```diff
@@ -157,6 +180,11 @@ class AssessmentForm extends Component {
         <Button onClick={this.handleSave} bsStyle="primary" bsSize="small">
           Save
         </Button>
+        {match.params.assessmentId ?
```
Comment (aljordan82) [inline gulp/src/partnerassessment/components/Form/AssessmentForm.jsx]: Should this be wrapped in UserCan?

**Comment by aljordan82** [comment]
@koonsies what's the advantage of confirm over the custom modal?

**Comment by koonsies** [comment]
I am a bit torn on that myself. For me, I was wondering what the advantage of a custom modal was over the built-in (as we have to maintain a custom, but built may be more reliable)

**Comment by koonsies** [comment]
I kind of lean toward the window.confirm, since that is what React-Router uses for Prompt, and we can't coerce another window type as of right now.

I don't have a major commitment to either. Kind of a toss up for me. Window.confirm was easier, less code overhead.

**Comment by aljordan82** [comment]
I think it would be better to keep the Alerts consistent across apps.

File: gulp/src/sass/partner_assessment/components/_assessment-form.scss [hunk 2]
```diff
@@ -30,17 +30,23 @@
...
+
   .form-group {
     label {
       margin-left: 5px;
       margin-bottom: 8px;
     }
   }
 
+  .btn {
+    margin-right: 5px;
```
Comment (aljordan82) [inline gulp/src/sass/partner_assessment/components/_assessment-form.scss]: The margin-right is affecting other buttons.

**Comment by aljordan82** [comment]
Front-end code is fine, this can be merged when the python code get reviewed

**Pull request merged**